src/App.tsx

```diff
@@ -1,30 +1,31 @@
 import './App.scss';
-
-/*
-import { RouteComponentProps } from 'react-router-dom';
-
-type TabsPageProps = React.FC<RouteComponentProps<{ tabId: string }>>;
-const TabsPage: TabsPageProps = ({ match }) => {...};
-
-or
-
-import { useParams } from 'react-router-dom';
-
-const TabsPage = () => {
-  const { tabId } = useParams<{ tabId: string }>();
-  ...
-};
-*/
-
-// const tabs = [
-//   { id: 'tab-1', title: 'Tab 1', content: 'Some text 1' },
-//   { id: 'tab-2', title: 'Tab 2', content: 'Some text 2' },
-//   { id: 'tab-3', title: 'Tab 3', content: 'Some text 3' },
-// ];
+import { NavLink, Route, Routes } from 'react-router-dom';
+import TabsPage from './components/TabsPage';
+import { Tab } from './react-app-env';
+import HomePage from './components/HomePage';
+import 'bulma/bulma.sass';
+
+const tabs: Tab[] = [
+  { id: 'tab-1', title: 'Tab 1', content: 'Some text 1' },
+  { id: 'tab-2', title: 'Tab 2', content: 'Some text 2' },
+  { id: 'tab-3', title: 'Tab 3', content: 'Some text 3' },
+];
 
 const App = () => (
-  <div className="App">
-    <h1>Tabs with router</h1>
+  <div className="App container is-widescreen">
+    <header className="header">
+      <nav>
+        <NavLink to="/">Home</NavLink>
+        <NavLink to="/tabs">Tabs</NavLink>
+      </nav>
+    </header>
+    <Routes>
+      <Route path="/" element={<HomePage />} />
+      <Route path="tabs" element={<TabsPage tabs={tabs} />}>
+        <Route path=":tabId" element={<TabsPage tabs={tabs} />} />
+      </Route>
```

Comment:
suggestion: it would be more consistent to use one more route with `index` property
```suggestion
      <Route path="/tabs">
        <Route index element={<TabsPage tabs={tabs} />} />
        <Route path=":tabId" element={<TabsPage tabs={tabs} />} />
      </Route>
```